### ISSUE TITLE
safely overriding file header fields

### ISSUE THREAD

**Issue by ok1-ok1** [Cinchoo/ChoETL.NACHA#42]
Hi
I have seen below code in your test project to override header values
```
 ChoActivator.Factory = (t, args) =>
            {
                if (t == typeof(ChoNACHAFileHeaderRecord))
                {
                    var header = new ChoNACHAFileHeaderRecord();
                    header.Initialize();

                    //Overrride any values here...
                    header.FileCreationDate = DateTime.Today.AddDays(100);
                    return header;
                }
                return null;
            };
```
If multiple files are being written with separate ChoNACHAWriters (separate ChoNACHAConfigurations) concurrently, **sometimes** generated files has wrong filecreationdates. One file had another files FileCreationDate. What do you think?

**Comment by Cinchoo** [comment]
Putting lock around this block preventing the issue??

**Comment by ok1-ok1** [comment]
I did not try but it seems not. It is rarely happening, difficult to test. I see that ChoActivator's Factory is static and it seems that this should be set globally. I was attempting to set header.FileCreationDate with a local value which caused the issue.

**Comment by Cinchoo** [comment]
Put a lock around it in base library. Released new package v1.2.1.10
